src/reducers/scrollStateHelper.js

```diff
@@ -239,21 +248,18 @@ function calculateRenderedRowRange(state, scrollAnchor) {
 
   // Loop to walk the viewport until we've touched enough rows to fill its height
   let rowIdx = startIdx;
-  while (totalHeight < viewportHeight && rowIdx < rowsCount && rowIdx >= 0) {
+  let endIdx = rowIdx;
+  while (totalHeight < availableHeight && rowIdx < rowsCount && rowIdx >= 0) {
     totalHeight += updateRowHeight(state, rowIdx);
+    endIdx = rowIdx;
     rowIdx += step;
   }
 
-  const endIdx = rowIdx - step;
   let firstRowOffset = firstOffset;
   if (lastIndex !== undefined) {
     // Calculate offset needed to position last row at bottom of viewport
     // This should be negative and represent how far the first row needs to be offscreen
-    if (rowIdx >= 0) {
-      firstRowOffset = viewportHeight - totalHeight;
-    } else {
-      firstRowOffset = 0;
-    }
+    firstRowOffset = Math.min(availableHeight - totalHeight, 0);
```

Comment:
Fixed so we compute the offset even when all rows are visible, in case the first is only partially visible.